### ISSUE TITLE
GPU memory size over 4GB

### ISSUE THREAD

**Issue by cyk0521** [gpgpu-sim/gpgpu-sim_distribution#71]
I trying to run a very large scale application which consumes over 4GB GPU memory on GPGPU-sim dev branch with GTX1080Ti config.

However, it doesn't work since GPU memory corruption (maybe overflow). So, I inspected source codes of GPGPU-sim and found "address_type" variable type at src/abstract_hardware_model.h.

src/abstract_hardware_model.h: line 70, typedef unsigned address_type;

I thought it may cause the memory overflow, I modified "unsigned" to "unsigned long long" for 64bit addressing, but it causes new problems with my GPU kernel. (My CUDA application works fine with native GPU GTX1080Ti)

Is it correct current version of GPGPU-sim has <4GB limitation? If so, how can I overcome it?

**Comment by abhi1212** [comment]
What type of application are you trying to run?

**Comment by abhi1212** [comment]
Which cfg file are you using?
How many layers execute on the simulator before you get an overflow error.

**Comment by cyk0521** [comment]
I'm using vgg-16.cfg and imagenet1k.data to validate an input image.
Here is my command line: 
./darknet classifier predict cfg/imagenet1k.data cfg/vgg-16.cfg vgg-16.weights data/eagle.jpg

I face overflow error before execution of neural network layers.. GPU memory contents were corrupted while loading VGGnet-16 weight file.

As my observation, no GPU kernel was executed before memory corruption.

**Comment by EdwarDu** [comment]
@cyk0521 sorry for hijacking the thread, but is it possible to share your port of darknet? We are trying to use darknet with gpgpu-sim also facing the same issue regarding cuBLAS and cuRAND. Though we are using tiny yolo without memory overflow issue (for now).

**Comment by abhi1212** [comment]
Yes, Even I would like to look at it once, have you posted it on git?

**Comment by gangmul12** [comment]
While running resnet50, i faced the same issue.
Due to the overflow of the variable that holds address value, functional simulation fails.

![capture](https://user-images.githubusercontent.com/11365962/61545682-fdadb180-aa82-11e9-90a9-480d96d937e9.PNG)
